ta4j-core/src/test/java/org/ta4j/core/XlsTestsUtils.java

```diff
@@ -46,11 +57,16 @@ this software and associated documentation files (the "Software"), to deal in
      * @param clazz class containing the file resources
      * @param fileName file name of the file containing the workbook
      * @return Sheet number zero from the workbook (mutable)
-     * @throws IOException if the workbook constructor or close throws
-     *             IOException
+     * @throws IOException if inputStream returned by
+     *             getResourceAsStream is null or if HSSFWorkBook constructor
+     *             throws IOException or if close throws IOException
      */
     private static Sheet getSheet(Class<?> clazz, String fileName) throws IOException {
-        HSSFWorkbook workbook = new HSSFWorkbook(clazz.getResourceAsStream(fileName));
+        InputStream inputStream = clazz.getResourceAsStream(fileName);
+        if (inputStream == null) {
```

Comment:
`null` check is not needed, because exactly the same will be done next line in the constructor of [HSSFWorkbook](https://poi.apache.org/apidocs/org/apache/poi/hssf/usermodel/HSSFWorkbook.html#HSSFWorkbook-java.io.InputStream-)

Reply:
Actually, the workbook constructor does not perform a null check.  If inputStream is null, it throws an NPE which isn't very helpful to the developer.  It only throws an IOException if the stream is non-null but cannot be ready.  Try renaming one of the XLS files without this check and you'll see an NPE.  Then add the check and you get a nice message stating that the file is not found.

Reply:
```
    public HSSFWorkbook(InputStream s, boolean preserveNodes)
            throws IOException
    {
        this(new NPOIFSFileSystem(s).getRoot(), preserveNodes);
    }
```

Reply:
Okay, just thought that this check was enough
```java
public static ReadableByteChannel newChannel(final InputStream in) {
        checkNotNull(in, "in");
       (...)
}
```
but the check is not reached